Content.Server/Blob/BlobTileSystem.cs

```diff
@@ -0,0 +1,528 @@
+using System.Linq;
+using System.Numerics;
+using Content.Server.Construction.Components;
+using Content.Server.Destructible;
+using Content.Server.Emp;
+using Content.Server.Flash;
+using Content.Server.Flash.Components;
+using Content.Shared.Blob;
+using Content.Shared.Damage;
+using Content.Shared.Destructible;
+using Content.Shared.FixedPoint;
+using Content.Shared.Popups;
+using Content.Shared.Verbs;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.GameStates;
+using Robust.Shared.Map;
+
+namespace Content.Server.Blob;
+
+public sealed class BlobTileSystem : SharedBlobTileSystem
+{
+    [Dependency] private readonly IMapManager _map = default!;
+    [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+    [Dependency] private readonly BlobCoreSystem _blobCoreSystem = default!;
+    [Dependency] private readonly AudioSystem _audioSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly EmpSystem _empSystem = default!;
+
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        // SubscribeLocalEvent<BlobTileComponent, ComponentStartup>(OnStartup);
+        SubscribeLocalEvent<BlobTileComponent, DestructionEventArgs>(OnDestruction);
+        SubscribeLocalEvent<BlobTileComponent, BlobTileGetPulseEvent>(OnPulsed);
+        SubscribeLocalEvent<BlobTileComponent, GetVerbsEvent<AlternativeVerb>>(AddUpgradeVerb);
+        SubscribeLocalEvent<BlobTileComponent, GetVerbsEvent<Verb>>(AddRemoveVerb);
+        SubscribeLocalEvent<BlobTileComponent, ComponentGetState>(OnGetState);
+        SubscribeLocalEvent<BlobTileComponent, FlashAttemptEvent>(OnFlashAttempt);
+    }
+
+    private void OnFlashAttempt(EntityUid uid, BlobTileComponent component, FlashAttemptEvent args)
+    {
+        if (args.Used == null || MetaData(args.Used.Value).EntityPrototype?.ID != "GrenadeFlashBang")
+            return;
+        if (component.BlobTileType == BlobTileType.Normal)
```

Comment:
This should be a discrete component that does it's own event handling, not a comparison against an enum.